src/app/compare/Compare.tsx

```diff
@@ -65,6 +66,8 @@ interface State {
   sortedHash?: string | number;
   sortBetterFirst: boolean;
   comparisonSets: { buttonLabel: React.ReactNode; items: DimItem[] }[];
+  adjustedPlugs?: { [itemId: string]: { [socketIndex: number]: DimPlug } } | undefined;
```

Comment:
Record assumes any requested values exist unless you `| undefined` them, 
`NodeJS.Dict` will automatically cause string type keys and possible undefined for any lookups, on dict-style objects.

`adjustedPlugs?: NodeJS.Dict<NodeJS.Dict<DimPlug>>;` is probably appropriate here,
going by the way the data is accessed later

might want to store
adjustedPlugs and adjustedStats as their own documented types because they are used in multiple files and need to always match

also the `?` after `adjustedPlugs` and `adjustedStats` (in various files)
makes the `| undefined` redundant

Reply:
I've made some temporary types - It's probably best if you just don't look at them for now, I'll check out `NodeJS.Dict` to replace them.